### PR TITLE
[evm] implement vectors in storage

### PR DESCRIPTION
**Pull request by emmazzz**
This PR finishes the implementation of vectors on EVM by adding support for vectors in storage: moving to/from storage, etc. Also some code refactoring is done.

### CODE REVIEW

File: language/evm/move-to-yul/src/functions.rs [hunk 1]
```diff
@@ -1000,50 +999,23 @@ impl<'a> FunctionGenerator<'a> {
...
             assert_eq!(byte_offs % 32, 0, "pointer fields are on word boundary");
-            if ty.is_vector() {
-                ctx.env
-                    .error(&self.parent.contract_loc, "vectors not yet implemented");
-                continue;
-            }
             ctx.emit_block(|| {
-                let hash = self.type_hash(ctx, ty);
-                // Add hash to names to avoid naming collision.
-                let linked_src_name = format!("$linked_src_{}", hash);
```
Comment (wrwg) [inline language/evm/move-to-yul/src/functions.rs:1011]: Is there a rebase for your previous PR missing?

**Review by wrwg — COMMENTED** [review]
Looks great. I'm wondering whether there is a rebase on previous PRs missing? Adding type hash to locals seems to be removed.

File: language/evm/move-to-yul/src/functions.rs (hunk 1 above)
Comment (emmazzz) [reply language/evm/move-to-yul/src/functions.rs:1011]: Ah I moved part of this to `create_and_move_data_to_linked_storage` function instead, to enable more code reuse.

**Review by wrwg — DISMISSED** [review]

**Pull request closed without merging**